framework/src/base/DisplacedProblem.C

```diff
@@ -573,8 +577,10 @@ DisplacedProblem::cacheResidualNeighbor(THREAD_ID tid)
 void
 DisplacedProblem::addCachedResidual(THREAD_ID tid)
 {
-  _assembly[tid]->addCachedResidual(_mproblem.residualVector(Moose::KT_TIME), Moose::KT_TIME);
-  _assembly[tid]->addCachedResidual(_mproblem.residualVector(Moose::KT_NONTIME), Moose::KT_NONTIME);
+  if (_nl->hasVector("Re_time"))
```

Comment:
Check with a string comparison. This is potentially expensive. Is this cachable as a `bool`?